x-pack/elastic-agent/pkg/artifact/install/hooks/hooks_installer.go

```diff
@@ -55,3 +62,34 @@ func (i *Installer) Install(programName, version, installDir string) error {
 
 	return nil
 }
+
+// Check performs installation check of program to ensure that it is already installed, then
+// runs the InstallerCheckSteps to ensure that the installation is valid.
+func (i *InstallerChecker) Check(programName, version, installDir string) error {
+	err := i.checker.Check(programName, version, installDir)
+	if err != nil {
+		return err
+	}
+
+	// installer check steps
+	nameLower := strings.ToLower(programName)
+	_, isSupported := program.SupportedMap[nameLower]
+	if !isSupported {
+		return nil
+	}
+
+	for _, spec := range program.Supported {
+		if strings.ToLower(spec.Name) != nameLower {
+			continue
+		}
+
+		if spec.CheckInstallSteps != nil {
+			return spec.CheckInstallSteps.Execute(installDir)
```

Comment:
can this also hang?

Reply:
If we implement a timeout on the `exec_file` step then no this will not hang. Without a timeout in `exec_file` then as a side-effect, this will hang here.